src/notes.c

```diff
@@ -329,7 +331,10 @@ static int note_new(
 		git_signature_dup(&note->committer, git_commit_committer(commit)) < 0)
 		return -1;
 
-	note->message = git__strndup(git_blob_rawcontent(blob), git_blob_rawsize(blob));
+	blobsize = git_blob_rawsize(blob);
+	GITERR_CHECK_BLOBSIZE(blobsize);
+
+	note->message = git__strndup(git_blob_rawcontent(blob), (size_t)blobsize);
```

Comment:
By the way, why do we even have this split between `git_off_t` and `size_t` when representing blobs? Shouldn't we uniformly use the same type everywhere in the first place? `git_blob` is an internal structure, too, so we are free to change it to use `size_t`.

Reply:
I'm always in favor of being explicit, so I'd be perfectly happy with e.g. `uint64_t`.

Reply:
I agree, this is noted in #4960